apm-sniffer/apm-sdk-plugin/rabbitmq-plugin/src/main/java/org/apache/skywalking/apm/plugin/rabbitmq/ChannelNConstructorInterceptor.java

```diff
@@ -21,12 +21,24 @@
 import com.rabbitmq.client.Connection;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.skywalking.apm.util.StringUtil;
 
 public class ChannelNConstructorInterceptor implements InstanceConstructorInterceptor {
     @Override
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
         Connection connection = (Connection) allArguments[0];
-        String url = connection.getAddress().toString().replace("/", "") + ":" + connection.getPort();
+
+        String[] parts = connection.getAddress().toString().split("/");
```

Comment:
Could you explain what do you mean `split` through `/`, then read <0> or <1>? The UT seems can't represent the use cases.

Reply:
For `InetAddress.getByName("localhost")`, the result of `InetAddress.toString()` is `localhost/127.0.0.1`.

For `InetAddress.getByName("127.0.0.1")`, the result of `InetAddress.toString()` is `/127.0.0.1`.

I also try `InetAddress.getHostName()` or `InetAddress.getHostAddress()`, but he former must return domain, while the latter must return IP.

So, I choose to split the result of `.toString()`.

Reply:
As a peer, I don't think we need to resolve anything, it doesn't matter. We just need an unique address(DNS, IP, alias name, domain name, anything) to represent a remote service.
That is all.

Reply:
For rabbitMQ case, it would be only used in `virtual MQ service` name on the UI. This should be the only place you could see the point of the data. At the kernel analysis level, it is used in network alias mapping. You could find this table/index in your storage.